packages/cli/src/cli/github/make.cog.pr.ts

```diff
@@ -0,0 +1,116 @@
+import { Github, owner, repo } from './github.js';
+
+export interface ImageryUrl {
+  tileMatrix: string;
+  url: string;
+}
+
+export class MakeCogGithub extends Github {
+  imagery: string;
+  constructor(imagery: string, logger: LogType) {
+    super(logger);
+    this.imagery = imagery;
+  }
+
+  /**
+   * Get the master aerial tileset config from the basemaps config repo
+   */
+  async getTileSetConfig(): Promise<ConfigTileSetRaster> {
+    this.logger.info({ imagery: this.imagery }, 'GitHub: Get the master TileSet config file');
+    const path = 'config/tileset/aerial.json';
+    const response = await this.octokit.rest.repos.getContent({
+      owner,
+      repo,
+      path,
+    });
+    if (!this.isOk(response.status)) throw new Error('Failed to get aerial TileSet config.');
+    if ('content' in response.data) {
+      return JSON.parse(Buffer.from(response.data.content, 'base64').toString());
+    } else {
+      throw new Error('Unable to find the content.');
+    }
+  }
+
+  /**
+   * Prepare and create pull request for the aerial tileset config
+   */
+  async createTileSetPullRequest(
+    layer: ConfigLayer,
+    urls: ImageryUrl[],
+    jira: string | undefined,
+  ): Promise<number | undefined> {
+    // Prepare new aerial tileset config
+    const tileSet = await this.getTileSetConfig();
+    const newTileSet = await this.prepareTileSetConfig(layer, tileSet);
+
+    // skip pull request if not an urban or rural imagery
+    if (newTileSet == null) return;
+
+    const branch = `feat/config-${this.imagery}`;
+    const ref = `refs/heads/${branch}`;
+    // Create branch first
+    const existing = await this.getBranch(branch);
+    if (existing == null) await this.createBranch(branch, ref);
+
+    // Create blob for the tileset config
+    const content = JSON.stringify(newTileSet, null, 2) + '\n'; // Add a new line at end to match the prettier.
+    const path = `config/tileset/aerial.json`;
+    const blob = await this.createBlobs(content, path);
+
+    // commit blobs to tree
+    const message = `feat(imagery): Add imagery ${this.imagery} config file.`;
+    await this.commit(branch, ref, [blob], message);
+    // Create imagery import pull request
+    const title = `feat(aerial): Config imagery ${this.imagery} into Aerial Map. ${jira ? jira : ''}`;
+    // Prepare pull request body
+    let body = `Imagery imported for ${this.imagery}, please use the following QA url once the aws job finished.\n\n`;
+    for (const url of urls) body += `Individual Imagery ${url.tileMatrix}: ${url.url}\n\n`;
+    const prNumber = await this.createPullRequest(branch, ref, title, body, true);
+    return prNumber;
+  }
+
+  /**
+   * Prepare aerial tileSet config json
+   */
+  async prepareTileSetConfig(
+    layer: ConfigLayer,
+    tileSet: ConfigTileSetRaster,
+  ): Promise<ConfigTileSetRaster | undefined> {
+    // Set layer zoom level and add to latest order
+    if (layer.name.includes('rural')) {
```

Comment:
if the layer names no longer have rural or urban in them, how does this work?

Can we pull this info from the stac somehow?

Reply:
Good point, This happens before when importing `manawatu-whanganui`. This is just create an empty draft PR with the correct title. We need manually edit the file and update the PR.

Reply:
This been fixed. We are adding the new layer at bottom with following configs.
 ```
layer.minZoom = 32;
layer.category = 'New Aerial Photos';
```